solutions/cses-1661.mdx

```diff
@@ -61,5 +66,57 @@ int main() {
 }
 
 ```
+	
+</CPPSection>
+	
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class SubarraySumsII {
+	public static void main (String[] args) throws IOException {
+		Kattio io = new Kattio();
+		int arraySize = io.nextInt();
+		int target = io.nextInt();
+		int[] array = new int[arraySize];
+
+		for (int x = 0; x < arraySize; x++) {
+			array[x] = io.nextInt();
+		}
+		
+		long prefixSum = 0;
+		long answer = 0;
+		Map<Long, Integer> sums = new HashMap<Long, Integer>();
+		sums.put((long) 0, 1);
```

Comment:
is `(long)` necessary

Reply:
yes according to my compiler, it always grays out if it isn't necessary but this time it is necessary and didn't gray out